src/database_connection.cpp

```diff
@@ -247,97 +305,41 @@ QueryResult DatabaseConnection::query(DatabaseQuery query) {
     }
     case QueryType::SELECT: {
       pqxx::result pqxx_result = performSelect(query);
-      QueryResult result{};
-      result.table = query.table;
-      auto count = query.fields.size();
-
-      for (auto row : pqxx_result) {
+      QueryResult result{.table=query.table};
```

Comment:
It's really around here where the relevant changes were made (removing explicit method definitions and replacing with template method)